src/cls/rbd/cls_rbd_client.cc

```diff
@@ -495,25 +495,29 @@ namespace librbd {
       for (vector<snapid_t>::const_iterator it = ids.begin();
            it != ids.end(); ++it) {
         snapid_t snap_id = it->val;
-        bufferlist bl1, bl2, bl3, bl4;
+        bufferlist bl1, bl2, bl3, bl4, bl5;
         ::encode(snap_id, bl1);
         op->exec("rbd", "get_snapshot_name", bl1);
-        ::encode(snap_id, bl2);
-        op->exec("rbd", "get_size", bl2);
+	::encode(snap_id, bl2);
+	op->exec("rbd", "get_snapshot_namespace", bl2);
```

Comment:
Since this cls method won't exist on older OSDs, unfortunately, until we get to the point of only supporting Kraken+, you need to retrieve the snapshot namespaces as part of an individual OSD op in case it errors out with -EOPNOTSUPP (which you could then ignore)